docs/troubleshooting.md

```diff
@@ -0,0 +1,75 @@
+# Troubleshooting
+
+The Collector offers multiple ways to measure the health of the Collector
+as well as investigate issues.
+
+- By default, Prometheus metrics are exposed on port `8888` and path
+`/metrics`. A grafana dashboard can be found
+[here](https://grafana.com/grafana/dashboards/11575).
+
+- Logs can be helpful in identifying issues. The verbosity level, which
+defaults to `INFO` can also be adjusted by passing the `--log-level` flag
+to the `otelcol` process. See `--help` for more details.
+
+- [Local exporters](https://github.com/open-telemetry/opentelemetry-collector/tree/master/exporter#local-exporters)
+can be configured to inspect the data being processed by the Collector.
+
+- The [health_check](https://github.com/open-telemetry/opentelemetry-collector/blob/master/extension/README.md#health_check)
+extension, which by default is on port `13133`, can be used to ensure
+the Collector is functioning properly.
+
+- The [zpages](https://github.com/open-telemetry/opentelemetry-collector/blob/master/extension/README.md#zpages)
+extension, which by default is on port `55679`, can be used to check RPC
+stats via `/debug/rpcz` as well as trace data via `/debug/tracez`.
+
+- The [pprof](https://github.com/open-telemetry/opentelemetry-collector/blob/master/extension/README.md#pprof)
+extension, which by default is on port `1777`, allows you to profile the
+Collector as it runs.
+
+## Common Issues
+
+### Collector restarts
+
+The Collector may restart if it is not
+[sized properly](https://github.com/open-telemetry/opentelemetry-collector/blob/master/docs/performance.md)
+or configured properly, for example a queue size configured higher than
+available memory. In most cases, restarts are due to memory
+pressure. To mitigate this, we recommend you pass the
+`--mem-ballast-size-mib` parameter to the `otelcol` process as well as
```

Comment:
This isn't really intended to solve this problem. It's mainly to improve the throughput by reducing GC cycles.

Reply:
Good point, removing and recommending the memory limiter processor only